routes/decoratorRequestBody.go

```diff
@@ -0,0 +1,133 @@
+package routes
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"os"
+	"reflect"
+
+	"github.com/trackit/jsonlog"
+
+	"github.com/trackit/trackit2/util/req"
+)
+
+type RequestBody struct {
+	Example interface{}
+}
+
+func (rb RequestBody) Decorate(h Handler) Handler {
+	h.Func = rb.getFunc(h.Func)
+	h.Documentation = rb.getDocumentation(h.Documentation)
+	return h
+}
+
+func (rb RequestBody) getFunc(hf HandlerFunc) HandlerFunc {
+	validate, err := req.CreateValidator(rb.Example)
+	var handleWithValidation func(http.ResponseWriter, *http.Request, Arguments, reflect.Value) (int, interface{})
+	if err != nil {
+		logger := jsonlog.DefaultLogger
+		logger.Error("Failed to build validator for type %T.", rb.Example)
+		os.Exit(1)
```

Comment:
If there is a `os.Exit(1)` shouldn't the logging be `Fatal` instead of just an `Error` ?

Reply:
`jsonlog` doesn’t have `Fatal` yet.

Reply:
Okay